### PR TITLE
community[minor]: Add ability to specify file encoding and json encoding for FileChatMessageHistory

### PR DESCRIPTION
**Pull request by yabooung**
Description:
Add UTF-8 encoding support

Issue:
Inability to properly handle characters from certain languages (e.g., Korean)

Fix:
Implement UTF-8 encoding in FileChatMessageHistory

### CODE REVIEW

File: libs/community/langchain_community/chat_message_histories/file.py [hunk 1]
```diff
@@ -20,21 +20,25 @@ def __init__(self, file_path: str) -> None:
         self.file_path = Path(file_path)
         if not self.file_path.exists():
             self.file_path.touch()
-            self.file_path.write_text(json.dumps([]))
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_message_histories/file.py:23]: write_text accepts an encoding parameter

write_text(json.dumps([]), encoding='utf-8')

File: libs/community/langchain_community/chat_message_histories/file.py [hunk 2]
```diff
@@ -20,21 +20,25 @@ def __init__(self, file_path: str) -> None:
...
+            with self.file_path.open("w", encoding="utf-8") as f:
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_message_histories/file.py:23]: To add the options in a non breaking way, you'll need to add encoding and ensure ascii as part of the initializer.

If we set the default encoding to utf-8, we could break existing user message history (e.g., if their default encoding is a windows encoding)

**Review by eyurtsev — CHANGES_REQUESTED** [review]
Thanks for the contribution! To merge, we need to expose the options via the initializer with defaults matching the previous defaults (encoding=None and ensure_ascii=True).

File: libs/community/langchain_community/chat_message_histories/file.py (hunk 2 above)
Comment (yabooung) [reply libs/community/langchain_community/chat_message_histories/file.py]: Thanks for the advice

**Review by Undertone0809 — APPROVED** [review]

File: libs/community/langchain_community/chat_message_histories/file.py [hunk 3]
```diff
@@ -11,30 +11,48 @@
 class FileChatMessageHistory(BaseChatMessageHistory):
     """Chat message history that stores history in a local file."""
 
-    def __init__(self, file_path: str) -> None:
+    def __init__(
+        self,
+        file_path: str,
```
Comment (eyurtsev) [inline libs/community/langchain_community/chat_message_histories/file.py:16]:
```suggestion
        file_path: str,
        *,
```

**Review by eyurtsev — APPROVED** [review]

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**